lib/src/summarize.dart

```diff
@@ -4,63 +4,302 @@
+    if (categories.containsKey(category)) {
+      List<String> returnSet = categories[category];
+      return returnSet.elementAt(randomizer % returnSet.length);
+    } else {
+      return null;
+    }
+  }
+
+  String _sentenceFiller(String word, [int size]) {
+    if (size != null) {
+      return _wordSelector(_categorySelector(word, size));
+    } else {
+      return _wordSelector(word);
+    }
+  }
+
+  String _additionList(List<String> list) {
+    if (list.length == 0) return '';
+    String englishList = ' Also, mentions ';
+    for (int i = 0; i < list.length; i++) {
+      englishList += list[i];
+      if (i < list.length - 2) {
+        englishList += ', ';
+      }
+      if (i == list.length - 2) {
+        if (i != 0) {
+          englishList += ',';
+        }
+        englishList += ' and ';
+      }
+    }
+    englishList += '. ';
+    return englishList;
+  }
+  
+  ///Testing add-ons
+  List<String> additionSearch() => _additionSearch();
+  
+  List<String> _additionSearch() {
+    List<String> features = new List<String>();
+    for (String feature in additionKeyWords.keys) {
+      if (dart.contains(feature)) {
+        features.add(additionKeyWords[feature]);
+      }
+    }
+    return features;
+  }
+
+  List<String> _codeSearch() {
+    List<String> features = new List<String>();
+    for (String feature in codeKeyWords.keys) {
+      if (dart.contains(feature)) {
+        features.add(codeKeyWords[feature]);
+      }
+    }
+    return features;
+  }
+
+  String _featureList(List<String> list) {
+    if (list.length == 0) return '. ';
+    String englishList = ', and ${_sentenceFiller('use')} use of ';
+    for (int i = 0; i < list.length; i++) {
+      englishList += list[i];
+      if (i < list.length - 2) {
+        englishList += ', ';
+      }
+      if (i == list.length - 2) {
+        if (i != 0) {
+          englishList += ',';
+        }
+        englishList += ' and ';
+      }
+    }
+    englishList += ' features. ';
+    return englishList;
+  }
+
+  String _packageList(List<String> list, {String source}) {
+    if (list.length == 0) {
+      if (source == null) return '';
+      else return '. ';
+    }
+    String englishList = '';
+    if (source == 'packages') englishList +=
+        ', and ${_sentenceFiller('code-0')} imports ';
+    else englishList += '${_sentenceFiller('code-1')} imports ';
+    for (int i = 0; i < list.length; i++) {
+      englishList += list[i];
+      if (i < list.length - 2) {
+        englishList += ', ';
+      }
+      if (i == list.length - 2) {
+        if (i != 0) {
+          englishList += ',';
+        }
+        englishList += ' and ';
+      }
+    }
+    if (source == 'packages') englishList += ' from external packages. ';
+    else englishList += ' from the dart package as well. ';
+    return englishList;
+  }
+
+  String _htmlCSS() {
+    String htmlCSS = 'This code has ';
+    if (html.length < 1) htmlCSS += 'no ';
```

Comment:
You may want to make this a `_hasHtml` method, so you can also check for null.

`bool get _hasHtml => html != null && html.isNotEmpty;`

ditto w/ css below